core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -548,8 +549,8 @@ private DefaultFunctionResolver unix_timestamp() {
   /**
    * WEEK(DATE[,mode]). return the week number for date.
    */
-  private DefaultFunctionResolver week() {
-    return define(BuiltinFunctionName.WEEK.getName(),
+  private DefaultFunctionResolver week(BuiltinFunctionName week) {
+    return define(week.getName(),
```

Comment:
Should accept `TIME` as an argument and thus depend on https://github.com/opensearch-project/sql/pull/1047

Reply:
```
mysql> select week(curtime(), 3);
+--------------------+
| week(curtime(), 3) |
+--------------------+
|                 47 |
+--------------------+
1 row in set (0.00 sec)
```

Reply:
Will be accomplished in future work
https://bitquill.atlassian.net/browse/AOS-1266